### DISCUSSION TITLE
`tabler_icon` function - `$default` use case ?

### DISCUSSION THREAD

**Discussion by cavasinf** [kevinpapst/TablerBundle#67] [General]
Hey @kevinpapst,

wondering what is the purpose of `$default` in the `tabler_icon` TWIG function.
https://github.com/kevinpapst/TablerBundle/blob/8a70eaa6d8d41b116803d0fe43a7f184210324c1/src/Twig/RuntimeExtension.php#L131-L134

Due to that, 
all of my calls of this function (when not using buttons), are made with `null` for that parameter.
```twig
{{ tabler_icon('email', null, false) }}
```

Two questions arise from this:
1. Should we use `false` to remove `.icon` class.
2. Should we override `.icon` css when using it outside of `.btn`.

My current problem is that the render with `icon` class and font-awesome is kinda ugly:
- Too big
- Not vertically centered
- Overlap with label when large icon (e.g : `fas fa-cogs`)

Today `.icon` class: 🥉 
![image](https://user-images.githubusercontent.com/25293190/156135674-cf9c14f4-2367-4d78-905b-264b54ee708c.png)

`.icon` class overridden: 🥈 
![image](https://user-images.githubusercontent.com/25293190/156135978-16199550-ed46-4e33-a1c4-7965136119ce.png)

Without `.icon` class: 🥇 
![image](https://user-images.githubusercontent.com/25293190/156136090-3d34d932-e197-4d3c-8f33-d62ee447fd27.png)

WDYT ?

**Comment by kevinpapst** [comment]
I have no such example in my app. 
Icons are exclusively for menus, buttons and expect from that just a handful of places where it really brings a benefit. 
Not judging, but why do you have an icon there? Does it help understanding the card/form?

Anyways: the default value is probably not necessary in 99% of all calls. I think it was mainly introduced for the menu, where you can add fontawesome classes directly (`fas fa-cogs` instead of `settings`). So If you don't have a mapping in tabler.yaml it will still show a menu icon. 
I needed that for my app, because plugins ship their own menu items and cannot rely on an icon mapping, but I cannot think of any other place where this is used.

To answer your question: IMO the order of the parameters is wrong, because you want to toggle the `$isIcon` without the need to define the default. And I could use that as well. 
Then we can use `{{ tabler_icon('email', false) }}` or `<i class="{{ 'email'|icon(false) }} text-green"></i>`.

Do you want to PR that change?

**Reply by cavasinf** [reply]
> but why do you have an icon there? Does it help understanding the card/form?

Yes it does, 
here is my full page:

![image](https://user-images.githubusercontent.com/25293190/156158342-bb8c6335-4ef4-478e-b508-73a672799bfe.png)

For those card titles I've removed `.icon` class.

----------------------------------

> IMO the order of the parameters is wrong

Agree

-----------------------------------

> Do you want to PR that change?

I can, but if you can too, go ahead 😃

**Comment by cavasinf** [comment]
TBH, `.icon` class only works for button if you are using FontAwesome.

Even for my menu, I've removed the `.icon` class.
As I've mentioned above,
> 

> render with `icon` class and font-awesome is kinda ugly:
> * Too big
> * Not vertically centered
> * Overlap with label when large icon (e.g : `fas fa-cogs`)

Example menu WITH and WITHOUT `.icon` class:

- With / Without
![image](https://user-images.githubusercontent.com/25293190/156159938-4b9c8123-9987-43c5-8c9c-4813f76428e2.png)![image](https://user-images.githubusercontent.com/25293190/156159399-130de982-e364-48a2-b3cf-07b837ae9bc9.png)

**Reply by kevinpapst** [reply]
Yes, in the menu there is no icon class. But in some other places, this here for example looks good. 
For demonstration I removed the `.icon` on the left side. The `<i>` is inside a `<a class="card-btn" ...>`.
This looks correct to me in both versions (but I know that it can look crappy in other places):
<img width="447" alt="Bildschirmfoto 2022-03-01 um 13 51 31" src="https://user-images.githubusercontent.com/533162/156172143-2c66a49b-3879-478d-be9e-4347f3338132.png">

If we change the argument order, we should rename `$isIcon` to `$isButton` to make the purpose clearer.
Not sure about your app, but mine has probably 50 places where the icon function is used and less than 10 are outside of a button.
So having the `icon` class by default made sense to me.
